src/Debug/Trace.purs

```diff
@@ -28,6 +28,22 @@ traceShow = traceAny <<< show
 -- | low-level debugging.
 foreign import traceAny :: forall a b. DebugWarning => a -> (Unit -> b) -> b
 
+-- | Call the JavaScript engine's debugger. There are a couple things to note
+-- | about this:
+-- |
+-- | - The debugger will open inside the `debugger` function, which probably
+-- |   isn't what you want. The call site will be the second entry in the call
+-- |   stack, and from there it'll work as expected.
+-- |
+-- | - This is not necessarily supported in all JavaScript engines. According
+-- |   to the spec, JavaScript engines without debugging functionality will
+-- |   treat this as a no-op.
+foreign import debugger :: forall a. DebugWarning => a -> a
+
+-- | Call `debugger` within an applicative context.
+debuggerA :: forall f a. DebugWarning => Applicative f => a -> f a
+debuggerA = pure <<< debugger
```

Comment:
The `A` actions turn out to be not a great idea, they often fire earlier than you want them too, so I'll be removing them in the updates for 0.12 anyway (in favour of `M` versions). 

Could we add this as an `M` in the first place? The implementation should be something like `debuggerM a = pure unit >>= \_ -> pure (debugger a)`. The bind is necessary to ensure that it will only evaluate `debugger` when evaluation proceeds within the chain of binds, rather than it firing immediately upon definition.

Reply:
Could perhaps make it `debuggerM :: forall m. DebugWarning => Monad m => m Unit` too? Or do you think feeding a value through is still useful in this context?

Reply:
Nah, I think you're right - will update :)